Bottom: 503cb42debb02a4139637566d6b699eafdb84bbf
Top:    d0f45a2bb0d4979efd0a1589fd079f77dd865535
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2010-12-09 23:08:43 +0000

Refresh of read-the-return-value-from-sca

---

diff --git a/src/hid/common/hidnogui.c b/src/hid/common/hidnogui.c
index 6111ec3..2b98a75 100644
--- a/src/hid/common/hidnogui.c
+++ b/src/hid/common/hidnogui.c
@@ -268,9 +268,13 @@ static int
 nogui_confirm_dialog (char *msg, ...)
 {
   int rv;
-  printf ("%s ? 0=cancel 1=ok : ", msg);
-  fflush (stdout);
-  scanf ("%d", &rv);
+
+  do
+    {
+      printf ("%s ? 0=cancel 1=ok : ", msg);
+      fflush (stdout);
+    }
+  while (scanf ("%d", &rv) != 1);
   return rv;
 }
